chromium_src/chrome/browser/banners/app_banner_manager_desktop_browsertest.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "src/chrome/browser/banners/app_banner_manager_desktop_browsertest.cc"
+
+#include "brave/components/constants/webui_url_constants.h"
+namespace webapps {
+
+class AppBannerManagerDesktopBrowserTestForBraveAllowedPage
+    : public AppBannerManagerDesktopBrowserTest {
+ public:
+  void SetUp() override {
+    scoped_feature_list_.InitWithFeatures(
+        /*enabled_features=*/{}, GetDisabledFeatures());
+    TestAppBannerManagerDesktop::SetUp();
+    AppBannerManagerBrowserTestBase::SetUp();
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(AppBannerManagerDesktopBrowserTestForBraveAllowedPage,
+                       WebUiAIChatApp) {
+  TestAppBannerManagerDesktop* manager =
+      TestAppBannerManagerDesktop::FromWebContents(
+          browser()->tab_strip_model()->GetActiveWebContents());
+
+  // Simulate loading an AI Chat page
+  {
+    base::RunLoop run_loop;
+    manager->PrepareDone(run_loop.QuitClosure());
+
+    ASSERT_TRUE(ui_test_utils::NavigateToURL(
+        browser(), GURL(base::StrCat({"chrome://", kAIChatUIHost}))));
+    run_loop.Run();
+  }
```

Comment:
I think this could be simplifed.
```
  ASSERT_TRUE(ui_test_utils::NavigateToURL(
      browser(), GURL(base::StrCat({"chrome://", kAIChatUIHost}))));
  manager->WaitForInstallableCheck();
```

Reply:
I'm not sure why the chromium browser test to check the same thing for chrome://password-manager doesn't do that

Reply:
But great spot - this works!

Reply:
pushed